### ISSUE TITLE
Patch Shuffle Operation

### ISSUE THREAD

**Issue by XiongZhongxia** [damo-cv/TransReID#66]
```python
    try:
        x = x.view(batchsize, group, -1, dim)
    except:
        x = torch.cat([x, x[:, -2:-1, :]], dim=1)
        x = x.view(batchsize, group, -1, dim)
```
Could you please help me understand how this part carry out shuffle operation? I thought there's just a concat...

**Comment by heshuting555** [comment]
This operation is just a 'group' operation. Shuffle operation is composed of 'group' and the subsequent 'transpose' operation.
